CMakeLists.txt

```diff
@@ -619,6 +625,19 @@ if(WITH_OPTIM)
             add_definitions(-DS390_DFLTCC_INFLATE)
             list(APPEND ZLIB_ARCH_SRCS ${ARCHDIR}/dfltcc_inflate.c)
         endif()
+        if(WITH_CRC32_VX)
+            add_definitions(-DS390_CRC32_VX)
+            set(VX_CRC32_SRCS ${ARCHDIR}/crc32-vx.c)
+            list(APPEND ZLIB_ARCH_SRCS ${VX_CRC32_SRCS})
+            set(VX_CRC32_FLAGS "-march=z13")
```

Comment:
I think the flags should go with the other compiler flag detection code. Notice how like `AVX2FLAG` it is set elsewhere in `cmake/detect-intrinsics.cmake` (see `check_avx2_intrinsics`). I think perhaps there should be a separate cmake function called `check_vx_crc32_intrinsics()` that would do this detection. 

The same thing should be done for configure because it has similar set of functions.